### DISCUSSION TITLE
Postgres + Nuxt Starter template gives 'missing_connection_string' when running local

### DISCUSSION THREAD

**Discussion by robertbel** [vercel/community#2304] [Help]
### Summary

I want to play around with the new storage features you guys have added recently, but can't get the Postgress one to work locally. I used [this template](https://vercel.com/templates/nuxt/postgres-nuxt) to start out. It deployed perfectly fine, but when i try to run it locally i'm not sure how to get the data from the database. This is what i have done so far:

- Cloned repo to my pc
- Run npm i
- Added a .env.local and filled it with the variables that are provided in the Quickstart snippets
- Run "npm run serve"
- 
Now i get the page with all the static stuff, but no data from the database. When i visit localhost:3000/api/get-users i get the following message:

`"VercelPostgresError - 'missing_connection_string': You did not supply a 'connectionString' and no 'POSTGRES_URL' env var was found."`

### Additional information

Not sure if i need to post this here, but this is the `get-users.ts`:

```
import { createPool, sql } from '@vercel/postgres';

async function seed() {
  const createTable = await sql`
    CREATE TABLE IF NOT EXISTS users (
      id SERIAL PRIMARY KEY,
      name VARCHAR(255) NOT NULL,
      email VARCHAR(255) UNIQUE NOT NULL,
      image VARCHAR(255),
      "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP
    );
    `;

  console.log(`Created "users" table`);

  const users = await Promise.all([
    sql`
          INSERT INTO users (name, email, image)
          VALUES ('Guillermo Rauch', 'rauchg@vercel.com', 'https://pbs.twimg.com/profile_images/1576257734810312704/ucxb4lHy_400x400.jpg')
          ON CONFLICT (email) DO NOTHING;
      `,
    sql`
          INSERT INTO users (name, email, image)
          VALUES ('Lee Robinson', 'lee@vercel.com', 'https://pbs.twimg.com/profile_images/1587647097670467584/adWRdqQ6_400x400.jpg')
          ON CONFLICT (email) DO NOTHING;
      `,
    sql`
          INSERT INTO users (name, email, image)
          VALUES ('Steven Tey', 'stey@vercel.com', 'https://pbs.twimg.com/profile_images/1506792347840888834/dS-r50Je_400x400.jpg')
          ON CONFLICT (email) DO NOTHING;
      `,
  ]);
  console.log(`Seeded ${users.length} users`);

  return {
    createTable,
    users,
  };
}
export default defineEventHandler(async () => {
  const startTime = Date.now();
  const db = createPool();
  try {
    const { rows: users } = await db.query('SELECT * FROM users');
    const duration = Date.now() - startTime;
    return {
      users: users,
      duration: duration
    };
  } catch (error) {
    // @ts-ignore
    if (error?.message === `relation "users" does not exist`) {
      console.log(
        "Table does not exist, creating and seeding it with dummy data now..."
      );
      // Table is not created yet
      await seed();
      const { rows: users } = await db.query('SELECT * FROM users');
      const duration = Date.now() - startTime;
      return {
        users: users,
        duration: duration
      };
    } else {
      throw error;
    }
  }
});
```

And because of the previously provide message about the connectionString, i thought i had to do something like this:

```
 const db = createPool({
    connectionString: process.env.POSTGRES_URL + "?sslmode=require",
  });
```

But this gives the following message:

`"VercelPostgresError - 'invalid_connection_string': This connection string is meant to be used with a direct connection. Make sure to use a pooled connection string or try createClient() instead."`

Not sure what that means. I have checked the [docs on errors](https://vercel.com/docs/errors/vercel-postgres-error-codes), but they didn't help me out. Also this [quickstart guide](https://vercel.com/docs/storage/vercel-postgres/quickstart) didn't really help me also.

Can someone here maybe help me out to get this to work locally?

### Example

https://postgres-nuxt.vercel.app/

**Comment by elliott-with-the-longest-name-on-github** [comment]
Can you report the results of this?:

```diff
import { createPool, sql } from '@vercel/postgres';

+console.log({
+  POSTGRES_URL: process.env.POSTGRES_URL,
+  POSTGRES_URL_NON_POOLING: process.env.POSTGRES_URL_NON_POOLING
+});

async function seed() {
  const createTable = await sql`
    CREATE TABLE IF NOT EXISTS users (
      id SERIAL PRIMARY KEY,
      name VARCHAR(255) NOT NULL,
      email VARCHAR(255) UNIQUE NOT NULL,
      image VARCHAR(255),
      "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP
    );
```

You don't have to share the full string logged (remove the identifying information) -- I just want to make sure the connection strings are available in your environment.

**Comment by elliott-with-the-longest-name-on-github** [comment]
I do also notice the template says to run `npm dev` and you said you're running `npm serve` -- was that a typo? (Just checking off all the "easy" boxes first!)

**Comment by robertbel** [comment]
@tcc-sejohnson  Hi! The env variables are not available there. I looked up some video's about `runtimeConfig` in nuxt.config file, and then i was able to access the variables with `useRuntimeConfig()`. So i thought i was on the right track until i  randomly tried to change the `.env.local` to just `.env` and this makes everything work with the code straight from the Vercel template.

Maybe i'll dive a little more in all this sometime soon, because i have no idea why this works and what that whole `runTimeConfig` stuff is for.

**Comment by daab** [comment]
Hi, Robertbel, any solution to achieve a correct connection string? Can you give me an example?

**Comment by AlaaHassoun** [comment]
Hi, make sure to change `.env.example` file to `.env`.

**Reply by kheder99** [reply]
thanks, that's work for me

**Comment by maxpaj** [comment]
My issue was accidentally importing the file where the SQL connection is set up in one of my client components.

**Comment by cyb3rsalih** [comment]
I've changed the name `.env.example` to `.env` and paste the data from vercel.  Then got same error. Guess what, I forgot to save the .env file  😂

**Comment by jaycoolslm** [comment]
I have had this issue in the past a few times with Svelte.

One solution I found was to create the pool client using the env variable and create sql queries from there:
```
const pool = createPool({
	connectionString: env.POSTGRES_URL
});
```

Another solution which I just discovered is to use `vercel dev` when running the app rather than `npm run dev`
